datascience/tables.py

```diff
@@ -1155,12 +1155,54 @@ def stack(self, key, labels=None):
         return type(self)([key, 'column', 'value']).with_rows(rows)
 
     def join(self, column_label, other, other_label=None):
-        """Generate a table with the columns of self and other, containing rows
+        """Generates a new table with the columns of self and other, containing rows
```

Comment:
nit: How about just "Create a new table"? I think that's more consistent with the language we use in other methods.